src/MagnumPlugins/StbVorbisImporter/StbVorbisImporter.cpp

```diff
@@ -0,0 +1,95 @@
+/*
+    This file is part of Magnum.
+
+    Copyright © 2010, 2011, 2012, 2013, 2014, 2015
+              Vladimír Vondruš <mosra@centrum.cz>
+    Copyright © 2015 Jonathan Hale <squareys@googlemail.com>
+
+    Permission is hereby granted, free of charge, to any person obtaining a
+    copy of this software and associated documentation files (the "Software"),
+    to deal in the Software without restriction, including without limitation
+    the rights to use, copy, modify, merge, publish, distribute, sublicense,
+    and/or sell copies of the Software, and to permit persons to whom the
+    Software is furnished to do so, subject to the following conditions:
+
+    The above copyright notice and this permission notice shall be included
+    in all copies or substantial portions of the Software.
+
+    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+    IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+    THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+    FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+    DEALINGS IN THE SOFTWARE.
+*/
+
+#include "StbVorbisImporter.h"
+
+#include <Corrade/Utility/Assert.h>
+#include <Corrade/Utility/Debug.h>
+#include <Corrade/Utility/Endianness.h>
+
+#include "stb_vorbis.c"
+
+namespace Magnum { namespace Audio {
+
+StbVorbisImporter::StbVorbisImporter() = default;
+
+StbVorbisImporter::StbVorbisImporter(PluginManager::AbstractManager& manager, std::string plugin): AbstractImporter(manager, std::move(plugin)) {}
+
+StbVorbisImporter::~StbVorbisImporter() { close(); }
+
+auto StbVorbisImporter::doFeatures() const -> Features { return Feature::OpenData; }
+
+bool StbVorbisImporter::doIsOpened() const { return _data; }
+
+void StbVorbisImporter::doOpenData(Containers::ArrayView<const char> data) {
+
+    Int numChannels, frequency;
+    Short* decodedData;
+
+    Int samples = stb_vorbis_decode_memory(reinterpret_cast<const UnsignedByte*>(data.data()), data.size(), &numChannels, &frequency, &decodedData);
+    _data = Containers::Array<char>(reinterpret_cast<char*>(decodedData), samples*numChannels*2);
```

Comment:
It will be done implicitly when the local Array instance goes out of scope, no?

If I remember correctly, memory in stb_* is allocated using `malloc` so you can't use `delete`. See e.g. fd9726c52aea146fb2ff93978c57678ed5ffb023 for solution. Consult the documentation for details (might be done differently in stb_vorbis, I don't know).